src/main/java/org/opensearch/neuralsearch/processor/factory/RerankProcessorFactory.java

```diff
@@ -40,6 +42,11 @@ public class RerankProcessorFactory implements Processor.Factory<SearchResponseP
     private final MLCommonsClientAccessor clientAccessor;
     private final ClusterService clusterService;
 
+    public RerankProcessorFactory(ClusterService clusterService) {
```

Comment:
There is already @AllArgsConstructor added at the top. You can use that while calling from test methods.
This constructor is not needed.

Reply:
I can see what you mean as it is already being set to null in the definition of the constructor. My intention was to not pass in a clientAccessor as a null parameter.

Reply:
Ive implemented your suggestion after thinking through some more. Since this is a factory it wont be as confusing to users who would like to create the processor since they will just create the processor using the factory.